### ISSUE TITLE
Don't use exec()

### ISSUE THREAD

**Issue by craigds** [globocom/derpconf#14]
I found this out by reading the source code - apparently a derpconf .conf file is actually a python file, and derpconf reads and executes it.

That's awfully insecure, given that local users might have access to write to the file. 

In addition the `conf_name` and `lookup_paths` options allow loading the config from a list of paths rather than a well-defined absolute path, meaning that all an attacker has to do is be able to write a file into *one* of the directories in the list to get arbitrary code execution.

cyclops uses this to load configs from `.` and `~` if a relative path is specified, which seems like a bad idea.

i'm not 100% sure on how to best solve this, but my 5minute 2-cents-worth: seems like most configs should be securely parseable using a very simple parser (read lines, split on `=`, then parse expressions using [ast.literal_eval()](https://docs.python.org/2/library/ast.html#ast.literal_eval) ). This would allow you to support a sane range of python expressions, without allowing arbitrary execution. It makes multi-line expressions a bit tricky though.

**Comment by heynemann** [comment]
It also kills the ability to use any python specific code (like getting the number of CPUs or using an environment variable or event an IF). Not sure how to solve code execution. Maybe not allowing the list of paths would improve security? I'm trying not to sacrifice extensibility and not compromise on security.

If we were to use just what you describe, we'd be better of using .ini files (natively supported by python). Maybe a good solution would be allowing users to specify which they want to use. INI files would not allow python code to run, but are more limited. Conf files are more extensible, with all the problems that come with it.

**Comment by craigds** [comment]
Fair enough. Maybe just suggesting (and generating) config files with a .py extension would be enough. Makes it really clear it's a python file and not some kind of INI type thing.
